src/components/Request/Pay/Views/Initial.view.tsx

```diff
@@ -97,26 +100,20 @@ export const InitialView = ({
             }
         }
 
-        const isXChain = selectedChainID !== requestLinkData.chainId
+        const isXChain = selectedTokenData?.chainId !== requestLinkData.chainId
             || !utils.areTokenAddressesEqual(
-                selectedTokenAddress,
+                selectedTokenData?.tokenAddress,
                 requestLinkData.tokenAddress
                 )
         setIsXChain(isXChain)
 
         // wait for token selector to fetch token price, both effects depend on
         // selectedTokenAddress and selectedChainID, but we depend on that
         // effect being completed first
-        if (!isConnected || (isXChain && !isTokenPriceFetchingComplete)) return
+        if (!isConnected || (isXChain && !selectedTokenData)) return
 
         estimateTxFee()
-    }, [
-        selectedTokenAddress,
-        selectedChainID,
-        selectedTokenDecimals,
-        isTokenPriceFetchingComplete,
-        requestLinkData
-    ])
+    }, [selectedTokenData, requestLinkData])
```

Comment:
Suggestion: should this not also depend on what address we're sending it from? ie connected wallet address?

Reply:
hmmm, we check for connection  so It probably make sense